### PR TITLE
[tests-only][full-ci] added test to update file using file id

### PR DESCRIPTION
**Pull request by PrajwolAmatya**
## Description
This PR adds the test scenarios for updating the content of a file inside personal space, project space, inside sub-folder, and shared files/folders.
The added test is done with and without `remote.php`, in the destination URL.
- `/remote.php/dav/spaces/<fileId>`
- `/dav/spaces/<fileId>`

## Related Issue

- https://github.com/owncloud/ocis/issues/6737

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [ ] Technical debt
- [x] Tests only (no source changes)

## Checklist:

- [ ] Code changes
- [ ] Unit tests added
- [x] Acceptance tests added
- [ ] Documentation ticket raised: <link>

### CODE REVIEW

**Review by SagarGi — COMMENTED** [review]
LGTM :+1:

File: tests/acceptance/features/bootstrap/FeatureContext.php [hunk 1]
```diff
@@ -1206,13 +1220,13 @@ public function userHasSentHTTPMethodToUrlWithPassword(string $user, string $ver
...
 	 * @param string|null $url
-	 * @param TableNode|null $body
+	 * @param TableNode|string|null $body
 	 * @param string|null $password
 	 *
 	 * @return void
 	 * @throws GuzzleException
 	 */
-	public function sendingToWithDirectUrl(?string $user, ?string $verb, ?string $url, ?TableNode $body, ?string $password = null): void {
+	public function sendingToWithDirectUrl(?string $user, ?string $verb, ?string $url, $body, ?string $password = null): void {
```
Comment (saw-jan) [inline tests/acceptance/features/bootstrap/FeatureContext.php:1229]:
```suggestion
	public function sendingToWithDirectUrl(?string $user, ?string $verb, ?string $url, $body = null, ?string $password = null): void {
```

File: tests/acceptance/features/bootstrap/FeatureContext.php [hunk 2]
```diff
@@ -1206,13 +1220,13 @@ public function userHasSentHTTPMethodToUrlWithPassword(string $user, string $ver
...
 	 * @param string|null $url
-	 * @param TableNode|null $body
+	 * @param TableNode|string|null $body
 	 * @param string|null $password
 	 *
 	 * @return void
 	 * @throws GuzzleException
 	 */
-	public function sendingToWithDirectUrl(?string $user, ?string $verb, ?string $url, ?TableNode $body, ?string $password = null): void {
+	public function sendingToWithDirectUrl(?string $user, ?string $verb, ?string $url, $body = null, ?string $password = null): void {
```
Comment (saw-jan) [inline tests/acceptance/features/bootstrap/FeatureContext.php:1229]: Could you do this in this function and do setResponse() from When steps only and and if used in Given do the required assertion there 
```diff
- $this->response = HttpRequestHelper::sendRequest(
-	$fullUrl,
-	$this->getStepLineRef(),
-	$verb,
-	$user,
-	$password,
-	$headers,
-	$bodyRows,
-	$config,
-	$cookies
- );
+ return HttpRequestHelper::sendRequest(
+	$fullUrl,
+	$this->getStepLineRef(),
+	$verb,
+	$user,
+	$password,
+	$headers,
+	$bodyRows,
+	$config,
+	$cookies
+ );
```

File: tests/acceptance/features/bootstrap/FeatureContext.php [hunk 3]
```diff
@@ -1206,13 +1220,13 @@ public function userHasSentHTTPMethodToUrlWithPassword(string $user, string $ver
...
-	 * @param TableNode|null $body
+	 * @param TableNode|string|null $body
 	 * @param string|null $password
 	 *
-	 * @return void
+	 * @return ResponseInterface
 	 * @throws GuzzleException
 	 */
-	public function sendingToWithDirectUrl(?string $user, ?string $verb, ?string $url, ?TableNode $body, ?string $password = null): void {
+	public function sendingToWithDirectUrl(?string $user, ?string $verb, ?string $url, $body = null, ?string $password = null): ResponseInterface {
```
Comment (saw-jan) [inline tests/acceptance/features/bootstrap/FeatureContext.php:1229]: missed this line
https://github.com/owncloud/ocis/blob/2e038038977fb857474591178cbaf80e2f13e879/tests/acceptance/features/bootstrap/FeatureContext.php#L1201

Comment (SwikritiT) [inline tests/acceptance/features/bootstrap/FeatureContext.php:1229]: I think we want all the parameters to have type.

Comment (SwikritiT) [reply tests/acceptance/features/bootstrap/FeatureContext.php:1229]: this is something we did a while ago https://github.com/owncloud/QA/issues/694, I don't know if `mixed` type is yet available in `php7.4` but let's not leave the `type of $body here` . can we maybe pre process the body and send the body in string format in this function maybe? But if the changes will be a lot we can do that in other PR's

**Review by saw-jan — APPROVED** [review]

**Pull request merged**